### ISSUE TITLE
Error's message is a null, Android

### ISSUE THREAD

**Issue by ShestakovaOlga** [SAP/gigya-flutter-plugin#4]
Hi,
We are using your plugin in a project and, when we tried to extract the error message on Android, we had a problem.

In MainActivity.kt, for implementation we added: 

```
val plugin: GigyaFlutterPlugin = flutterEngine.plugins.get(GigyaFlutterPlugin::class.java) as GigyaFlutterPlugin
 plugin.registerWith(application, GigyaAccount::class.java, flutterEngine.dartExecutor.binaryMessenger)

```

And in AndroidManifest.xml file we have this:

```
       <meta-data
            android:name="apiKey"
            android:value="3_-MkajnAgr6HgZsps35bl_aYNMG338riz8FjDzDiBMRhzYw3wswmZQwyuzOubo2z4" />

        <meta-data
            android:name="apiDomain"
            android:value="us1.gigya.com" />

        <meta-data
            android:name="accountCacheTime"
            android:value="1" />

        <meta-data
            android:name="sessionVerificationInterval"
            android:value="60" />

```
The problem is that the error message has all values null

![image](https://user-images.githubusercontent.com/39344982/109009279-a46e2100-76ae-11eb-821b-c3e27339a8a8.png)

We also have tried the solution in the other open issue, but the mapped is also null. On IOS everything works fine, including the mapped object.

Do you have any solution or any idea why this is happening?

Thanks

**Comment by tal-mi** [comment]
Hi,

What call are you trying to make?
If u can please provide as much info as you can so I can replicate the issue and provide a quick solution.

Thanks,
Tal.

**Comment by ShestakovaOlga** [comment]
Hi,
We are making login and register requests, and we copied the implementation from the example folder of this repo. 

```
void onLogin(user, password) async {
    setState(() {
      _inProgress = true;
    });
    GigyaSdk.instance.login(user, password).then((result) {
      final response = Account.fromJson(result);
      setState(() {
        showError = false;
        _inProgress = false;
        _requestResult = 'Login success:\n\n ${response.uid}';
      });
    }).catchError((error) {
      setState(() {
        _inProgress = false;
        _requestResult = '${error.errorDetails}';
        showError = true;
      });
    });
  }

```
Thanks

**Comment by tal-mi** [comment]
Hi,

Thanks for reporting this.
I have found inconsistencies in our Android/iOS wrappers. 
We will fix that and issue an update soon.

As a workaround you can update the "decodeError" method in the "gigya_flutter_plugin.dat" file to the following:

mixin DataMixin {
  /// Mapping communication error structure.
  Map<String, dynamic> decodeError(PlatformException error) {
    if (error.details == null) return {};
    if (Platform.isAndroid) {
      final mapped = jsonDecode(error.details);
      return mapped;
    }
    final mapped = error.details.cast<String, dynamic>();
    return mapped;
  }
}

Regards,
Tal

**Comment by ShestakovaOlga** [comment]
Hi Tal,
I have updated the "decodeError" method and it works good for Android.
Thank you and we will be waiting for new updates.

**Comment by tal-mi** [comment]
Hi, 

Issue resolved in 0.0.3 release.

Thanks,
Tal
